test/test_runner/test_parser.cpp

```diff
@@ -226,9 +226,6 @@ std::string TestParser::extractTextBeforeNextStatement(bool ignoreLineBreak) {
 
 TestStatement* TestParser::extractStatement(TestStatement* statement,
     const std::string& testCaseName) {
-    if (endOfFile()) {
```

Comment:
I see endofFile() is not used anywhere else, let's remove that function as well.
@ray6080 I am a little confused why we check `endOfFile()` before, do you have thoughts there? Is it safe to remove?

Reply:
Isn't this to prevent reading over the end of the file?
I'm fine refactoring the logic here to check the return value of `nextLine();`